clients/elrond/transactionHandler.go

```diff
@@ -0,0 +1,86 @@
+package elrond
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+
+	crypto "github.com/ElrondNetwork/elrond-go-crypto"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/builders"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/core"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/data"
+)
+
+type transactionHandler struct {
+	proxy                   ElrondProxy
+	relayerAddress          core.AddressHandler
+	multisigAddressAsBech32 string
+	nonceTxHandler          NonceTransactionsHandler
+	relayerPrivateKey       crypto.PrivateKey
+	singleSigner            crypto.SingleSigner
+}
+
+// sendTransactionReturningHash will try to assemble a transaction, sign it, send it and, if everything is OK, returns the transaction's hash
+func (txHandler *transactionHandler) sendTransactionReturningHash(ctx context.Context, builder builders.TxDataBuilder, cost uint64) (string, error) {
+	tx, err := txHandler.signTransaction(ctx, builder, cost)
+	if err != nil {
+		return "", err
+	}
+
+	return txHandler.nonceTxHandler.SendTransaction(context.Background(), tx)
+}
+
+func (txHandler *transactionHandler) signTransaction(ctx context.Context, builder builders.TxDataBuilder, cost uint64) (*data.Transaction, error) {
```

Comment:
rename cost to gasLimit

Reply:
the cost is gasLimit * gasPrice

Reply:
right, copied from the old implementation